packages/astro/components/ViewTransitions.astro

```diff
@@ -25,13 +25,14 @@ const { fallback = 'animate' } = Astro.props as Props;
 	// you can figure it using an index. On pushState the index is incremented so you
 	// can use that to determine popstate if going forward or back.
 	let currentHistoryIndex = history.state?.index || 0;
-	if (!history.state) {
+	if (!history.state && transitionEnabledOnThisPage()) {
 		persistState({ index: currentHistoryIndex, scrollY: 0 });
 	}
 
 	const supportsViewTransitions = !!document.startViewTransition;
-	const transitionEnabledOnThisPage = () =>
-		!!document.querySelector('[name="astro-view-transitions-enabled"]');
+	function transitionEnabledOnThisPage() {
```

Comment:
went with arrow functions to save bytes, can we change this back?

Reply:
Sure ;-)
I had the choice to move the code up, or make it a function. Will revert.

Reply:
Done. This time, even the checks passed :)